app/Models/Allocation.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Models;
+
+use App\Models\Common;
+use App\Models\Batch;
+
+class Allocation extends Common
+{
+    protected $table = 'UserBatch';
+    public $timestamps = false;
+    const CREATED_AT = 'added_on';
+    protected $fillable = ['user_id','batch_id','level_id','role'];
+
+    public function batch(){
+      return $this->beiongsToMany('App\Models\Batch');
+    }
+
+    public function teacher(){
+      return $this->belongsToMany('App\Models\User')->where('UserBatch.role','teacher');
+    }
+
+    public function mentor(){
+      return $this->belongsToMany('App\Models\User')->where('UserBatch.role','mentor');
+    }
+
+    public function level(){
+      return $this->belongsToMany('App\Level');
+    }
+
+    public function get($batch_id, $user_id,$role="teacher",$level_id=0){
```

Comment:
Don't use this function name. And, ideally I'd like to getTeacher() and getMentor() calls, with a baseGet()

Reply:
get() is part of laravel Eloquent - so might cause issues.